### ISSUE TITLE
Syntax of target-specific annotations

### ISSUE THREAD

**Issue by liujed** [p4lang/p4-spec#695]
Compiler implementations are free to interpret the semantics of any annotation not defined by the spec, but the syntax of these annotations is currently constrained to be either an `expressionList` or a `keyValueList`. Why not give annotation implementers free reign over their syntax? Something like:
```
annotation
  : '@' name
  | '@' name '(' annotationBody ')'
  ;

annotationBody
  : expressionList
  | keyValueList
  | freeFormAnnotationBody
  ;
```
where `freeFormAnnotationBody` is any sequence of tokens containing balanced parentheses.

Currently, annotation implementers can use strings to achieve a similar effect (e.g., `@name("foo bar baz")`), but it would be nice if they didn't have to, especially if the body of the annotation refers to components of the P4 program.

**Comment by liujed** [comment]
Here is an implementation of this idea: https://github.com/p4lang/p4c/pull/1614

**Comment by mbudiu-vmw** [comment]
Please write a PR against the spec spelling out the new rules for free-form annotations.

**Comment by liujed** [comment]
Addressed in #724. Closing.
